src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityQuantumChest.java

```diff
@@ -161,23 +184,53 @@ public boolean hasFrontFacing() {
     public NBTTagCompound writeToNBT(NBTTagCompound data) {
         NBTTagCompound tagCompound = super.writeToNBT(data);
         if (!itemStack.isEmpty() && itemsStoredInside > 0L) {
-            tagCompound.setTag("ItemStack", itemStack.writeToNBT(new NBTTagCompound()));
-            tagCompound.setLong("ItemAmount", itemsStoredInside);
+            tagCompound.setTag(NBT_ITEMSTACK, itemStack.writeToNBT(new NBTTagCompound()));
+            tagCompound.setLong(NBT_ITEMCOUNT, itemsStoredInside);
         }
         return tagCompound;
     }
 
     @Override
     public void readFromNBT(NBTTagCompound data) {
         super.readFromNBT(data);
-        if (data.hasKey("ItemStack", NBT.TAG_COMPOUND)) {
-            this.itemStack = new ItemStack(data.getCompoundTag("ItemStack"));
+        if (data.hasKey(NBT_ITEMSTACK, NBT.TAG_COMPOUND)) {
+            this.itemStack = new ItemStack(data.getCompoundTag(NBT_ITEMSTACK));
             if (!itemStack.isEmpty()) {
-                this.itemsStoredInside = data.getLong("ItemAmount");
+                this.itemsStoredInside = data.getLong(NBT_ITEMCOUNT);
+            }
+        }
+    }
+
+    @Override
+    public void initFromItemStackData(NBTTagCompound itemStack) {
+        super.initFromItemStackData(itemStack);
+        if (itemStack.hasKey(NBT_ITEMSTACK, NBT.TAG_COMPOUND)) {
+            this.itemStack = new ItemStack(itemStack.getCompoundTag(NBT_ITEMSTACK));
+            if (!this.itemStack.isEmpty()) {
+                this.itemsStoredInside = itemStack.getLong(NBT_ITEMCOUNT);
             }
+        } else if (itemStack.hasKey(NBT_PARTIALSTACK, NBT.TAG_COMPOUND)) {
+            importItems.setStackInSlot(0, new ItemStack(itemStack.getCompoundTag(NBT_PARTIALSTACK)));
         }
     }
 
+    @Override
+    public void writeItemStackData(NBTTagCompound itemStack) {
+        super.writeItemStackData(itemStack);
+        if (!this.itemStack.isEmpty()) {
+            itemStack.setTag(NBT_ITEMSTACK, this.itemStack.writeToNBT(new NBTTagCompound()));
+            itemStack.setLong(NBT_ITEMCOUNT, itemsStoredInside + this.itemStack.getMaxStackSize());
+        } else {
+            ItemStack partialStack = exportItems.extractItem(0, 64, false);
+            if (!partialStack.isEmpty()) {
+                itemStack.setTag(NBT_PARTIALSTACK, partialStack.writeToNBT(new NBTTagCompound()));
+            }
+        }
```

Comment:
The different virtualization issue could be solved by serializing both the partial stack and any virtualized items, so that you don't have to make the assumption that the virtualized item and the partial stack are the same item.

Reply:
Instead of doing this, since I felt like an item getting virtualized "under" another (i.e., one item in output slot, none virtualized, and a different item is put into input and gets virtualized) was a confusing mechanic, I instead stopped this from happening. Now if the item in the output stack does not match the item in the input stack, the input item cannot be virtualized.